packages/hub/src/hub.ts

```diff
@@ -426,9 +427,9 @@ export class Hub implements HubInterface {
 
     if (scope) {
       // End existing session if there's one
-      const currentSession = scope.getSession && scope.getSession();
+      const currentSession = scope.getSession && (scope.getSession() as Session);
```

Comment:
When is `scope.getSession` ever undefined?

Reply:
Good point. TBH I don't know (the definedness check was there before I started refactoring). I checked and couldn't find another occurance of this which is why I'm gonna remove it for now (also the cast seems unnecessary).

Reply:
I just checked and it seems like this was introduced in #3017. So the way I understand it, this was introduced to fix a bug when people were using different versions of SDK packages(?). Given that the versions in question are quite old, I'm not entirely sure if we still need this check. WDYT @AbhiPrasad ?

Reply:
cc @kamilogorek

Reply:
It was breaking for some people when we introduced the new method, thus the check. But as you already noticed, especially with the major release, we should be safe to remove it.

Reply:
Thanks, removed in https://github.com/getsentry/sentry-javascript/pull/5054/commits/f1f4d2008fceedb740dc822a6e3e373711c56a62